pkg/secrets-store/identityserver.go

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package secretsstore
+
+import (
+	"github.com/container-storage-interface/spec/lib/go/csi"
+	csicommon "github.com/deislabs/secrets-store-csi-driver/pkg/csi-common"
+	"golang.org/x/net/context"
+
+	wrappers "github.com/golang/protobuf/ptypes/wrappers"
+)
+
+type identityServer struct {
+	*csicommon.DefaultIdentityServer
+}
+
+// Probe check whether the plugin is running or not.
+// This method does not need to return anything.
+// Currently the spec does not dictate what you should return either.
+// Hence, return an empty response
+func (ids *identityServer) Probe(ctx context.Context, req *csi.ProbeRequest) (*csi.ProbeResponse, error) {
+	return &csi.ProbeResponse{Ready: &wrappers.BoolValue{Value: true}}, nil
```

Comment:
The comment and the implementation are different. This is always returning ready. Is that right?

Reply:
I'll update the comment. The `liveness-probe` sidecar container uses the Probe RPC call to check if ready is true. Currently we don't need any other criteria other than successfully able to make Probe RPC call for liveness probe. That's the reason it always returns true for ready.

Reply:
Updated the comment.